src/nifs/protogalaxy/poly/folded_witness.rs

```diff
@@ -109,24 +109,33 @@ fn fold_witnesses<F: PrimeField>(
             polys_L_in_challenges.len()
         ]),
     };
+
+    const CHUNK_SIZE: usize = 2usize.pow(10);
```

Comment:
Does the chunk_size depend on cpu threads? If that's the case, we should configure it in terms num_cpus